str/coloc/gwas_ld_runner.py

```diff
@@ -30,74 +31,150 @@
-    correlation_df['locus'] = correlation_df.index
-
-    # drop the STR locus from the list of SNPs (it will automatically have a correlation of 1)
-    correlation_df = correlation_df[correlation_df['locus'] != str_locus]
-
-    # add some attributes
-    correlation_df['gene'] = gene
-    correlation_df['str_locus'] = str_locus
-    correlation_df['celltype'] = celltype
-
-    # return the df as a String
-    return correlation_df.to_csv(index=False)
+    # read in gwas catalog file
+    gwas_catalog_orig = pd.read_csv(gwas_file)
+    # read in gene_annotation_table
+    gene_annotation_table = pd.read_csv(gene_annotation_file)
+
+    # obtain snp cis-window coordinates for the gene
+    gene_table = gene_annotation_table[gene_annotation_table['gene_ids'] == gene]  # subset to particular ENSG ID
+    start_snp_window = float(gene_table['start'].astype(float)) - 100000  # +-100kB window around gene
+    end_snp_window = float(gene_table['end'].astype(float)) + 100000  # +-100kB window around gene
+    chrom = gene_table['chr'].iloc[0][3:]
+    print('Obtained SNP window coordinates')
+
+    # subset the gwas catalog to the snp_window
+    gwas_catalog = gwas_catalog_orig[gwas_catalog_orig['CHR'] == int(chrom)]
+    gwas_catalog = gwas_catalog[gwas_catalog['BP'] >= start_snp_window]
+    gwas_catalog = gwas_catalog[gwas_catalog['BP'] <= end_snp_window]
+
+    if gwas_catalog.empty:
+        print('No SNP GWAS data for ' + gene + ' in the cis-window: skipping....')
+        return
+
+    # obtain lead SNP (lowest p-value) in the snp_window of the GWAS
+    lowest_p_row = gwas_catalog.loc[gwas_catalog['P'].idxmin()]
+    lead_snp_chr = lowest_p_row['CHR']
+    lead_snp_bp = lowest_p_row['BP']
+    lead_snp_end = lowest_p_row['BP'] + 1
+    lead_snp_locus = f'{lead_snp_chr}:{lead_snp_bp}-{lead_snp_end}'
+
+    # load in the str fdr file
+    str_fdr_file = f'{str_fdr_dir}/{celltype}_qval.tsv'
+    str_fdr = pd.read_csv(str_fdr_file, sep='\t')
+
+    # obtain top STR locus for the gene (if multiple are tied - iterate over each)
+    str_fdr_gene = str_fdr[str_fdr['gene_name'] == gene]
+    for estr in zip(
+        ast.literal_eval(str_fdr_gene['chr'].iloc[0]),
+        ast.literal_eval(str_fdr_gene['pos'].iloc[0]),
+    ):
+        chr_num = estr[0][3:]
+        pos = estr[1]
+        end = str(int(pos) + 1)
+        str_locus = f'{chr_num}:{pos}-{end}'
+        write_path = output_path(
+            f'gwas_ld/{phenotype}/{celltype}/{gene}_chr{chr_num}_{pos}_gwas_ld_results.csv',
+            'analysis',
+        )
+
+        if to_path(write_path).exists():
+            print(f'GWAS LD for {gene} and {str_locus} already exists. Skipping...')
+            continue
+
+        print(f'Running LD for {gene} and {str_locus}')
+        snp_vcf_path = f'{snp_vcf_dir}/chr{chr_num}_common_variants.vcf.bgz'
+        str_vcf_path = f'{str_vcf_dir}/hail_filtered_chr{chr_num}.vcf.bgz'
```

Comment:
the nesting here means that you re-copy this per-chromosome VCF for every gene? Instead of copying to a generic 'local' name, can you copy it to 'local_{chr}_STR/SNP.vcf', and only copy a new one in if you haven't already copied that down? 

I may be misunderstanding the looping, but I'm assuming that the same chromosome can appear multiple times on line 92

Reply:
moved the local paths before the for loop